### PR TITLE
Add delete folder flow

### PR DESCRIPTION
**Pull request by CrystalPea**
### Work done:

- Add a delete folder link to manage folder page
- Add delete_folder method to template folder api client
- Add delete folder GET/POST endpoint 
- Add Flash messages and error handling to the airport
- Tests

Pivotal ticket: https://www.pivotaltracker.com/story/show/161177455

#### Happy path:

screen shot 2018-11-14 at 17 03 37

screen shot 2018-11-14 at 17 56 26

screen shot 2018-11-14 at 17 04 34

#### Folder not empty:

screen shot 2018-11-14 at 17 00 31

### CODE REVIEW

File: app/main/views/templates.py [hunk 1]
```diff
@@ -387,9 +391,45 @@ def manage_template_folder(service_id, template_folder_id):
...
+    if request.method == 'POST':
+        try:
+            template_folder_api_client.delete_template_folder(current_service.id, template_folder_id)
+
+            flash("'{}' folder has been removed".format(template_folder_name), 'info')
+            return redirect(
+                url_for('.choose_template', service_id=service_id)
+            )
+        except HTTPError as e:
+            msg = "Folder is not empty"
```
Comment (servingUpAces) [inline app/main/views/templates.py]: We can check that the folder is empty before calling delete.

File: app/templates/views/templates/manage-template-folder.html [hunk 2]
```diff
@@ -25,7 +25,15 @@ <h1 class="heading-large">
...
     {{ textbox(form.name) }}
-    {{ page_footer('Save') }}
+    {{ page_footer(
+      'Save',
+      delete_link=url_for(
+        '.delete_template_folder',
+        service_id=current_service_id,
+        template_folder_id=template_folder_id
+      ),
+      delete_link_text="Delete this folder") }}
```
Comment (servingUpAces) [inline app/templates/views/templates/manage-template-folder.html]: Do we want to only show the delete link if the folder is empty?

Comment (CrystalPea) [reply app/templates/views/templates/manage-template-folder.html]: We could, and then some kind of message if it's not? Like "If you want to delete this folder, you need to first empty it"...?

File: app/main/views/templates.py (hunk 1 above)
Comment (CrystalPea) [reply app/main/views/templates.py]: You're right 👍

File: app/main/views/templates.py [hunk 3]
```diff
@@ -387,6 +391,51 @@ def manage_template_folder(service_id, template_folder_id):
...
+        flash("'{}' folder is not empty".format(template_folder_name), 'info')
+        return redirect(url_for(
+            '.manage_template_folder', service_id, template_folder_id
+        ))
+
+    if request.method == 'POST':
+        try:
+            template_folder_api_client.delete_template_folder(current_service.id, template_folder_id)
+
+            flash("'{}' folder has been removed".format(template_folder_name), 'info')
```
Comment (servingUpAces) [inline app/main/views/templates.py]: We don't need this message

Comment (karlchillmaid) [reply app/main/views/templates.py]: `Are you sure you want to delete the [folder name] folder?`
`Confirm`

– as discussed, no need for a confirmation message –

Just finalising the _can't delete_ message

**Comment by karlchillmaid** [comment]
Can't delete message:

`You must empty this folder before you can delete it`
`Move or delete all templates and folders from the [folder name] folder and try again.`

**Comment by CrystalPea** [comment]
So, I changed the content of some messages and got rid of final confirmation message. I updated the screenshots to reflect those changes.

In a separate pull request I will change delete template 'are you sure' message so it is consistent with delete folder message, as per consultation with Karl.

@karlchillmaid @yahoopete @servingUpAces are you happy with how it all looks now?

**Comment by karlchillmaid** [comment]
@CrystalPea Content in the screenshots matches my previous comments and our discussion yesterday.

Good to know the _delete template_ flow will change to match this.

One change – please remove the quote marks around the folder name. We don't use quote marks around template names in the _delete template_ flow.

Not for now – perhaps we can revisit the option to add an extra description to the _can't delete_ message sometime?

File: app/templates/views/templates/manage-template-folder.html (hunk 2 above)
Comment (servingUpAces) [reply app/templates/views/templates/manage-template-folder.html]: We've decided it's better to show the button and let them get the `must be empty` message.

**Pull request merged**